### ISSUE TITLE
Remove `Clone` impl for `Api` (maybe?)

### ISSUE THREAD

**Issue by LukasKalbertodt** [telegram-rs/telegram-bot#37]
I'm wondering why one should clone an `Api`. `hyper::Client`  does not implement `Clone`, implementing it for `Api` introduces unnecessary complexity (especially with http timeouts). I also want to discourage users from cloning the `Api`, because it doesn't make sense (right?).

Sadly removing something like this is a breaking change :disappointed: 

@knsd any thoughts on this?

**Comment by knsd** [comment]
Actually I use `Clone` trait for `Api` fairly often. Something like this:

```rust
[derive(Clone)]
struct Event {
     api: Api,
     ...
}

impl Event {
    fn reply(...) {...}
    ...
}
```
So it's very easy to reply from several threads, and I'm sure it is very popular pattern.
But of course it's possible to store token instead of entire `Api` struct and I agree with `discourage users from cloning the Api`.
I've just created breaking change PR #38, so I think we could just release 0.5 version.

**Comment by LukasKalbertodt** [comment]
As the `Api` should implement `Sync`, I think the idiomatic way is to use `Arc<Api>` or just `&Api`. Both of those type are clonable and should work with your code above. And with `Arc` you won't have problems with lifetimes, if you can't prove to the compiler that your `Api` indeed lives long enough.

After all, `Client::new` could be costly. We'd also had to save the timeout in the `Api` as well, because we can't access it through the `Client` (I'd say this is a "bug" in hyper, though).

So I'm in favor of removing the `Clone` impl and maybe provide an example about `Arc` usage.

About breaking change: We could release your change now already or we could wait until we fixed #35  and release both changes together. What do you think?
